### DISCUSSION TITLE
Isometric Adjustment for Flare Engine.

### DISCUSSION THREAD

**Discussion by BuzzLEquerre** [flareteam/flare-game#978] [General]
Hello guys !
I'm having a problem with isometric angles for my blender houses. I started creating houses for my Flare game, only once in Tiled or in the game I encounter a slight rotation of my house and I don't know how to find the right setting : 
In game,
![Capture d'écran 2024-11-14 224505](https://github.com/user-attachments/assets/d4ce13f4-beeb-429b-8d4b-ee1ed647bb1b)
My camera setting in blender,
![Capture d'écran 2024-11-14 224656](https://github.com/user-attachments/assets/e14fb800-4f8f-4525-833c-c492f909b0b6)
![Capture d'écran 2024-11-14 224703](https://github.com/user-attachments/assets/9e9029f5-4e04-40d4-afb2-74376d08264d)

And my tilesetdefs : 
[tileset]
img=images/tilesets/city_house2.png

tile=3012,0,0,960,540,32,526

(the house)
![city_house2](https://github.com/user-attachments/assets/32c48a36-11e7-4210-bf9f-69f62a3d635a)

**Comment by dorkster** [comment]
Forgive me, but I'm not seeing the problem here. You describe a "slight rotation", but the edge of the house walls appear to be parallel to the edges between the grass and cobblestone tiles.

**Reply by BuzzLEquerre** [reply]
Yes that's what I think too, only once in the game the character switches behind the house a little after the middle of the house. So I suppose a tiny rotation but I can't solve this bug,
![Capture d'écran 2024-11-16 193639](https://github.com/user-attachments/assets/9b689c45-4e0e-4256-8009-b527d4450e18)
![Capture d'écran 2024-11-16 193647](https://github.com/user-attachments/assets/fd3cc9a4-0213-4f25-9af3-0654c656640b)

**Reply by dorkster** [reply]
Ah, I see now. This is because of the way that the draw order is done. Because the building an the player are both on the `object` layer, we determine which should be drawn in front based on the origin positions of each object. However, for most large objects, there's no place to put the origin that will make it behave correctly. There are sometimes exceptions, like the "rotten tower" set piece in flare-game. That one happens to work because the footprint is square.

But your building is not square, so what's the solution? You'll need to split the object into pieces that fit on a single tile in the world. For an example, have a look at the `tall structures` tiles on any of the grassland maps.

![Screenshot_20241117_124558](https://github.com/user-attachments/assets/8051f6cd-2559-4323-9a2d-07b4f7dd2cc4)

These tiles are meant to be placed on the map in an "L" shape (or "V" depending on how you want to look at it) to create the full object. Notice that all the tiles that *aren't* the one closest to the camera only use half the tile width. This is because the tiles closer to the camera will overlap that region, so it needs to be accounted for.

For flare-game, I think @clintbellanger did this by hand for the few large objects we have. If that's the case, I think creating a tool to do this process for a given image + tile size could be valuable. Because even after you figure out what to do, it's very tedious to do it by hand.

**Reply by BuzzLEquerre** [reply]
Okay, luckily I have photoshop, it's going to be difficult but it's doable, if I succeed I'll share the result with you.
And if there is a method to do it automatically I would like to know how to do it :)

**Reply by dorkster** [reply]
Success! I made a Python script to do the slicing process: https://gist.github.com/dorkster/326468b19b7cb7dc71948987fe4f70c1

I only did one manual edit to your image, which was to trim off the empty space on the left/right sides, while still keeping the width divisible by 1/2 the tile width (i.e. 32 pixels). 448 pixels wide works for your house:

![house](https://github.com/user-attachments/assets/b9e0a2f3-fc86-441b-8228-4904ab6a04e3)

I then ran the script like so:

```
./split_large_tile.py --input_file=./house.png --corner_x=160
```

`corner_x` is the x position of the left side of the tile closest to the camera. Here's the result:

![output](https://github.com/user-attachments/assets/bc71a581-b915-4542-b5c1-af8c32bcf273)

You should be able to use this in Tiled. Make a new tileset with it and set the tile size to 64x540.

**Reply by BuzzLEquerre** [reply]
I'm sorry but I don't know anything about python, I tried your program on Thonny and I don't seem to have the "PIl" module, how can I integrate it into your program ?
![Capture d'écran 2024-11-17 214423](https://github.com/user-attachments/assets/75ff6a6e-fd70-4e3f-b589-b69532fc6ebc)

**Reply by dorkster** [reply]
1. First, save the Python file where the `house.png` image is located (or copy the house to wherever you save the Python script).
2. In Thonny, go to **Tools >> Open system shell...**
3. In the window that opens, enter `pip install pillow` and press Enter. Close the window when it finishes installing.
4. Go to **View >> Program Arguments**. A text field should appear in the toolbar.
5. Enter `--input_file=./house.png --corner_x=160` into the program arguments text field.
6. Go to **Run >> Current script in terminal**. If it works, it will create an `output.png` file with the split tiles
